app/ui.js

```diff
@@ -1057,10 +1181,14 @@ const UI = {
         UI.rfb.addEventListener("securityfailure", UI.securityFailed);
         UI.rfb.addEventListener("capabilities", UI.updatePowerButton);
         UI.rfb.addEventListener("clipboard", UI.clipboardReceive);
-        document//.getElementById("noVNC_container")
-            .addEventListener('mouseenter', UI.enterVNC);
-        document//.getElementById("noVNC_container")
-            .addEventListener('mouseleave', UI.leaveVNC);
+
+        document.addEventListener('mouseenter', UI.enterVNC);
+	document.addEventListener('mouseleave', UI.leaveVNC);
+	document.addEventListener('blur', UI.blurVNC);
+	document.addEventListener('focus', UI.focusVNC);
+	document.addEventListener('focusout', UI.focusoutVNC);
+	document.addEventListener('mousemove', UI.mouseMoveVNC);
+	document.addEventListener('mousedown', UI.mouseDownVNC);
```

Comment:
Why do we need all of these events? Isn't it sufficient with just `focus`?

Reply:
I discovered while implementing the clipboard feature that the Chrome API clipboard.readText() is still buggy. It fails sometimes (not-deterministic why) with undefined error. When using all these events here to try to read it, it mostly works. I reported the bug to the Chrome community and suggest not implement this feature until it doesn't work properly.

Reply:
That's disappointing. Do you have a link?

Reply:
Yes of course, follow this: https://bugs.chromium.org/p/chromium/issues/detail?id=949926&q=clipboard&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified